server.go

```diff
@@ -553,7 +555,15 @@ func (srv *Server) serveDTLSConnection(ctx *shutdownContext, conn *coapNet.Conn)
 		// We will block poller wait loop when
 		// all pool workers are busy.
 		c := ClientConn{commander: &ClientCommander{session}}
-		srv.spawnWorker(&Request{Client: &c, Msg: msg, Ctx: sessCtx, Sequence: c.Sequence()})
+
+		// Try to set the peer certificates, but don't error if it fails as there
+		// might not be any peer certificates.
+		dtlsConn := conn.Connection().(*dtls.Conn)
+		cert := dtlsConn.RemoteCertificate()
+		flatCerts := bytes.Join(cert, nil)
+		certs, _ := x509.ParseCertificates(flatCerts)
```

Comment:
This parts can be implemented in function newSessionDTLS

Reply:
I've got the forwarding part via the commander implemeneted now! This part should be moved to the new `(s *sessionDTLS) RemoteCertificates()` function, right? If I move this to `newSessionDTLS`, then we'd need to store the certificates in the `sessionDTLS`.

That might be even nicer, what do you think?
1. The `RemoteCertificates` function gets and parses the certificates. It would becomes something like this, but with added error checking on the certificate.

   ```go
	func (s *sessionDTLS) RemoteCertificates() ([]*x509.Certificate, error) {
		dtlsConn := s.connection.Connection().(*dtls.Conn)
		cert := dtlsConn.RemoteCertificate()
		flatCerts := bytes.Join(cert, nil)
		certs, _ := x509.ParseCertificates(flatCerts)
		return certs, nil
	}

   ```
2. `newSessionDTLS` gets and parses the certificates and stores them in `sessionDTLS`. This way the `RemoteCertificates` just becomes a 'getter` function. and wouldn't need to return an error. I'd be adding the code to get the certificates [here](https://github.com/go-ocf/go-coap/blob/master/sessiondtls.go#L32), and return an error from `newSessionDTLS` in case of an parse error.

Reply:
I've just pushed af98d1a, where I went for option 2 since I think that would be the 'cleanest' solution. If you, however, rather see it another way, please let me know and I'll update it.